File: frontend/lib/features/study/models/lesson_content.dart
class LessonContent {
  final String nodeId;
  final String title;
  final String content;

  LessonContent({
    required this.nodeId,
    required this.title,
    required this.content,
  });

  factory LessonContent.fromJson(Map<String, dynamic> json) {
    return LessonContent(
      nodeId: json['nodeId'],
      title: json['title'],
      content: json['content'],
    );
  }
}

File: backend/src/main/java/com/SeeTohJJ/Backend/study/model/content/LessonContent.java
package com.SeeTohJJ.Backend.study.model.content;

import com.SeeTohJJ.Backend.study.model.StudyNode;

import java.time.LocalDateTime;

public class LessonContent extends StudyNode {

    private String nodeId;
    private String title;
    private String content;
    private LocalDateTime lastUpdated;

    public String getNodeId() {
        return nodeId;
    }

    public void setNodeId(String nodeId) {
        this.nodeId = nodeId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public LocalDateTime getLastUpdated() {
        return lastUpdated;
    }

    public void setLastUpdated(LocalDateTime lastUpdated) {
        this.lastUpdated = lastUpdated;
    }
}
